Task3/Task3_Atanas_Kisyov/src/python/car_wash/db_connection/pool_base_class.py

```diff
@@ -0,0 +1,25 @@
+"""Abstract base class to ensure that database pools implement the correct methods"""
+import psycopg2
+
+from python.car_wash.utils.singleton import Singleton
+
+
+class PoolBaseClass(metaclass=Singleton):
+
+    """Abstract base class for creating connection pools"""
+
+    def __enter__(self) -> psycopg2.connect:
```

Comment:
Based on my last version of example you will not have enter and exit

Reply:
Done.